the-collab-lab/tcl-37-smart-shopping-list
# Read from and write to the Firestore database so that users can persist information

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description



In this PR, we created two new files: `firebase.js` and `List.js`. In `firebase.js` we added the config information to connect firebase and firestore to the app. In `List.js` is where we displayed the items of our database and sent new data to the database.

In `firebase.js` we found that using `import firebase from "firebase"` conflicts with `import { initializeApp } from 'firebase/app'` so we used `import { getFirestore } from 'firebase/firestore'`  to connect to the database instead. 

In `List.js` we learned that we have to manually add the document id when pushing the document data to an array. 


## Related Issue
closes #1 



## Acceptance Criteria


- [x] firebase has been added as a project dependency
- [X] Making a change in the Firestore database will update the data in the app
- [X] Making a change in the app will update the data in Firestore


## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
| ✓ | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before


<img width="1400" alt="Screen Shot 2022-01-12 at 4 06 04 PM" src="https://user-images.githubusercontent.com/67841237/149221611-8ce9e0e8-6409-48fb-a03a-ab0d91d2c765.png">


### After
<img width="1400" alt="Screen Shot 2022-01-12 at 4 08 31 PM" src="https://user-images.githubusercontent.com/67841237/149221785-a6bbca6c-a5fe-4aa4-bb06-69e74b726676.png">





## Testing Steps / QA Criteria
1. npm install
2. npm start
3. You should expect to see the collection data in the UI on the site
4.  To send data to Firestore, click the "Send data to database" button.
5. UI is updated with the data that is hard coded into sendData()
